src/util-debounce.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Delay the execution of a function until a timer has elapsed.
+ * Repeated calls within the time period resets the timer.
+ * @param timeout A minimum time between calls in milliseconds.
+ * @param callback A function to call.
+ * @example
+ * ```ts
+ * import storeHook from "tyin/hook";
+ * import debounce from "tyin/util-debounce";
+ *
+ * const useExample = storeHook({ a: 1, b: 2 });
+ *
+ * useExample.subscribe(debounce(5000, (_, newState) => {
+ *   fetch("/draft", { method: "PUT", body: JSON.stringify(newState) });
+ * }));
+ * ```
+ */
+export default function debounce<T extends (...args: any[]) => void>(
+  timeout: number,
+  callback: T
```

Comment:
nit: Callback is usually before timeout in similar functions, in e.g. `setTimeout` .

Reply:
Yeah, I know. It's just, a bit... yuk. I think it makes the most sense to reason "Delay for X, then do Y", so the args should come in that order. It also generally formats better. But I will reconsider before merging, thanks for the feedback. :)

Reply:
> Yeah, I know. It's just, a bit... yuk. I think it makes the most sense to reason "Delay for X, then do Y", so the args should come in that order. It also generally formats better. But I will reconsider before merging, thanks for the feedback. :)

Copied your deleted comment from the mail.

I agree that the order looks nicer with the number first. Callbacks tend also to span several lines which makes the timeout argument easy to miss. The only argument for having `timeout` last is to be inline with JavaScript APIs, but we can agree that that is not the best role model.  As long as this is consistent within the library, I think your style is better. 

I got a deja vu of this conversation, I think we had it before, and the same arguments came up. :smile: